src/content/en/fundamentals/security/credential-management/retrieve-credentials.md

```diff
@@ -4,369 +4,396 @@ book_path: /web/fundamentals/_book.yaml
-      </td>
-    </tr>
-    <tr>
-      <td>
-        <code>name</code>
-      </td>
-      <td>
-        <code>String</code><br>
-        User name string.
-      </td>
-    </tr>
-    <tr>
-      <td>
-        <code>iconURL</code>
-      </td>
-      <td>
-        <code>String</code><br>
-        User icon image url string.
-      </td>
-    </tr>
-  </tbody>
-</table>
-
-#### Change parameters
-
-In some cases, it may be necessary to add additional data to the
-authentication POST.
-
-Change the param keys by assigning a string to `.idName` or `.passwordName`.
-
-You can also add extra parameters such as a cross-site request forgery (CSRF)
-token by assigning `.additionalData` to the `FormData` and append 
-key-values to it.
-
-Once you get the credential object:
-
-    if (cred) {
-      if (cred.type == 'password') {
-        // Use `email` instead of `id` for the id
-        cred.idName = 'email';
-
-        // Append CSRF Token
-        var csrf_token = document.querySelector('#csrf_token').value;
-        var form = new FormData();
-        form.append('csrf_token', csrf_token);
-
-        // Append additional credential data to `.additionalData`
-        cred.additionalData = form;
-
-        // `POST` the credential object.
-        // id, password and the additional data will be encoded and
-        // sent to the url as the HTTP body.
-        fetch(url, {           // Make sure the URL is HTTPS
-          method: 'POST',      // Use POST
-          credentials: cred    // Add the password credential object
-        }).then(function() {
-          // continuation
-        });
-      }
-    }
+To implement federated login:
 
-You can do a similar thing by assigning a `URLSearchParams` object instead of
-a `FormData` to `.additionalData`. In this case, the whole credential object 
-is encoded using `application/x-www-form-urlencoded`.
+1. Authenticate the user with a third-party identity.
+2. Store the identity information.
+3. [Update UI or proceed to a personalized page](#update_ui) (same as auto sign-in).
 
-### Authenticate with an identity provider
+### Authenticate user with third-party identity
 
-To authenticate the user with an identity provider, simply use the 
-specific authentication flow with the `FederatedCredential`.
+When a user taps on a federated login button,
+run the specific identity provider authentication flow with the `FederatedCredential`.
 
 For example, if the provider is Google, use the
-[Google Sign-In JavaScript library](/identity/sign-in/web/):
+[Google Sign-In JavaScript library](https://developers.google.com/identity/sign-in/web/):
 
     // Instantiate an auth object
     var auth2 = gapi.auth2.getAuthInstance();
 
     // Is this user already signed in?
     if (auth2.isSignedIn.get()) {
       var googleUser = auth2.currentUser.get();
-      
+
       // Same user as in the credential object?
       if (googleUser.getBasicProfile().getEmail() === id) {
```

Comment:
The user id you could obtain from retrieved credential object. You may want to wrap this code example with something like:

```
   navigator.credentials.get({
     password: true,
     federated: {
       provider: [
         'https://accounts.google.com'
       ]
     },
     unmediated: false
   }).then(c => {
```

Then instead of just `id`, you could use `c.id`.